### PR TITLE
MS-140: Sessions Redux Implement

### PR DESCRIPTION
**Pull request by julianv97**

### CODE REVIEW

File: src/constants/actionTypes.js [hunk 1]
```diff
@@ -1 +1,19 @@
...
+export const ADD_SESSIONS_REJECTED = 'ADD_CLIENTS_REJECTED';
+export const DELETE_SESSIONS_FETCHING = 'DELETE_CLIENTS_FETCHING';
+export const DELETE_SESSIONS_FULFILLED = 'DELETE_CLIENTS_FULFILLED';
+export const DELETE_SESSIONS_REJECTED = 'DELETE_CLIENTS_REJECTED';
+export const UPDATE_SESSIONS_FETCHING = 'UPDATE_CLIENTS_FETCHING';
+export const UPDATE_SESSIONS_FULFILLED = 'UPDATE_CLIENTS_FULFILLED';
+export const UPDATE_SESSIONS_REJECTED = 'UPDATE_CLIENTS_REJECTED';
+export const GET_ONE_SESSION_FETCHING = 'GET_ONE_CLIENTS_FETCHING';
+export const GET_ONE_SESSION_FULFILLED = 'GET_ONE_CLIENTS_FULFILLED';
+export const GET_ONE_SESSION_REJECTED = 'GET_ONE_CLIENTS_REJECTED';
```
Comment (AndresCespedes23) [inline src/constants/actionTypes.js:16]: Shouldn't be 'GET_ONE_**SESSION**_REJECTED' the string of the const ?

**Review by AndresCespedes23 — APPROVED** [review]

File: src/Components/Sessions/index.js [hunk 2]
```diff
@@ -1,154 +1,86 @@
...
   const [idActive, setIdActive] = useState('');
-  const [showMessage, setShowMessage] = useState(false);
-  const [messageType, setMessageType] = useState('');
-  const [message, setMessage] = useState('');
-  const [isLoading, setLoading] = useState(false);
+  const sessions = useSelector((state) => state.sessions.list);
+  const isLoading = useSelector((state) => state.sessions.isLoading);
+  const showMessage = useSelector((state) => state.sessions.showMessage);
+  const message = useSelector((state) => state.sessions.messageText);
+  const messageType = useSelector((store) => store.sessions.messageType);
```
Comment (sofpag) [inline src/Components/Sessions/index.js:22]: why store insted of state? That breaks the consistency

File: src/Components/Sessions/index.js [hunk 3]
```diff
@@ -1,154 +1,86 @@
...
-      })
-      .catch((err) => console.log(err))
-      .finally(() => setLoading(false));
-  };
+  useEffect(() => {
+    dispatch(getSessions());
+  }, [dispatch]);
 
-  const cleanMessage = () => {
+  /* const cleanMessage = () => {
```
Comment (sofpag) [inline src/Components/Sessions/index.js:31]: maybe you should delete the comment
There are many in this file

File: src/redux/Sessions/thunks.js [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+});
+
+const getSessionsRejected = () => ({
+  type: GET_SESSIONS_REJECTED
+});
+
+export const getSessions = () => {
+  return (dispatch) => {
+    dispatch(getSessionsFetching());
+    fetch(BASE_URL)
```
Comment (sofpag) [inline src/redux/Sessions/thunks.js:37]: As I said in Slack, maybe we could give more information about the status when we use the fetch

**Review by sofpag — APPROVED** [review]

**Pull request merged**